### DISCUSSION TITLE
Loading custom dictionaries from a remote source (e.g. from a GitHub Gist)

### DISCUSSION THREAD

**Discussion by PaulRBerg** [streetsidesoftware/vscode-spell-checker#1857] [Q&A] [answered]
Hi, I love this VSCode extension, been using it for 3+ years now. My only problem is the large list of custom words that have accumulated under the `cSpell.userWords` field in my global VSCode settings.

Now, I know that there is an option to create a local custom dictionary on disk - I did read the [How to add your own Dictionaries](https://github.com/streetsidesoftware/vscode-spell-checker#global-dictionary) settings. But I would like to know if it would be possible to load the dictionary from a remote source, such as a GitHub Gist? I would prefer not to have to remember to transfer my custom dictionary when I switch to a new laptop.

**Comment by pcuci** [comment]
Hi @PaulRBerg—great idea!

My temporary workaround is to rely on: https://github.com/streetsidesoftware/vscode-spell-checker?tab=readme-ov-file#define-the-dictionary and store the custom dictionaries in my [dotfiles](https://github.com/pcuci/dotfiles/blob/main/.vscode/settings.json). Since I take these with me all over the place anyway it's almost as convenient as your idea.

Bonus: I can add new terms directly from any of my projects in any of the custom dictionaries—which I don't know how you'd achieve with a gist(?)

![image](https://github.com/user-attachments/assets/47660baf-1e8d-4a9a-afe9-e8e0545f2660)

**Reply by PaulRBerg** [reply]
Thanks for your feedback — I ended up doing something similar. I created a separate repo for keeping track of my custom dictionary. I need to remember to store the dictionary at the same path on all devices, but it works for now.

**Comment by Jason3S** [comment]
Support for remote configuration was added in CSpell 8 and the v4 of this extension.

See: [CSpell Gist Based Config](https://gist.github.com/Jason3S/6a88fd2993f4139a567787f4e5e19f40).

As @pcuci points out, the extension cannot support adding words to the remote word list.

**Comment by Jason3S** [comment, marked as answer]
# Gist Based CSpell Config

This is an example of a CSpell configuration stored in a [gist](https://gist.github.com/Jason3S/6a88fd2993f4139a567787f4e5e19f40).

There are two opitons:

1. Just having the dictionary in a gist.
2. Having the entire configuration in the gist.

## Using just a dictionary file from Gist

1. Get the raw url of dictionary file.
2. Reference the dictionary file in the `dictionaryDefinitions` of the CSpell Config.

**`cspell.cofnig.yaml`**
```ymal
dictionaryDefinitions:
  - name: gist-dict
    path: https://gist.githubusercontent.com/Jason3S/6a88fd2993f4139a567787f4e5e19f40/raw/dictionary.txt
dictionaries:
  - gist-dict
```

## Using the configuration from Gist

1. Get the raw url of the configuration file stored in the Gist.
2. Add the url to the `import` section of the cspell configuration.

**`cspell.config.yaml`**
```yaml
import:
  - https://gist.githubusercontent.com/Jason3S/6a88fd2993f4139a567787f4e5e19f40/raw/cspell.config.yaml
```
